### PR TITLE
Update default values in CoordinatorDynamicConfig

### PR DESCRIPTION
**Pull request by kfaraz**
## Changes

The defaults of the following config values in the `CoordinatorDynamicConfig` are being updated.

#### 1. `maxSegmentsInNodeLoadingQueue = 500` (previous = 100)

Rationale: With round-robin segment assignment now being the default assignment technique, the Coordinator can assign a large number of under-replicated/unavailable segments very quickly. Before round-robin, a large queue size would cause the Coordinato to get stuck in `RunRules` duty due to very slow strategy-based cost computations.

#### 2. `replicationThrottleLimit = 500` (previous = 10)
Rationale: Along with the reasoning given for `maxSegmentsInNodeLoadingQueue`, a very low `replicationThrottleLimit` can cause clusters to be very slow in getting to full replication, even when there are loading threads sitting idle.

Note: It is okay to keep this value equal to `maxSegmentsInNodeLoadingQueue`. Even with equal values, load queues will not get filled up with just replicas, and segments that are completely unavailable will still get a fair chance. This is because while MSINLQ applies to a single server, `replicationThrottleLimit` applies to each tier.

#### 3. `maxSegmentsToMove = 100` (previous = 5)

Rationale: A very low value of this config (say 5) turns out to be very ineffective in balancing especially if there are a large number of segments in a cluster and/or a large skew between usages of two historical servers.
On the other hand, a very large value can cause excessive moves every minute, which might have the following disadvantages:
- Load of moving segments competing with load of unavailable/under-replicated segments
- Unnecessary network costs due to constant download and delete of segments

These defaults will be revisited after #13197 is merged.

## Testing

These values have been tried on different production cluster sizes, and have been found to give satisfactory results.

#### Release note
Update default values of the following coordinator dynamic configs:
- `maxSegmentsInNodeLoadingQueue = 500`
- `maxSegmentsToMove = 100`
- `replicationThrottleLimit = 500`

<hr>

This PR has:

- [ ] been self-reviewed.
   - [ ] using the [concurrency checklist](https://github.com/apache/druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)
- [ ] added documentation for new or modified features or behaviors.
- [x] a release note entry in the PR description.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/dev/license.md)
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [x] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [ ] added integration tests.
- [x] been tested in a test Druid cluster.

### CODE REVIEW

**Comment by AmatyaAvadhanula** [comment]
Changes may be needed in `coordinator-dynamic-config.tsx` as well.

**Review by AmatyaAvadhanula — APPROVED** [review]

File: docs/configuration/index.md [hunk 1]
```diff
@@ -951,16 +951,16 @@ Issuing a GET request at the same URL will return the spec that is currently in
...
-|`maxSegmentsInNodeLoadingQueue`|The maximum number of segments that could be queued for loading to any given server. This parameter could be used to speed up segments loading process, especially if there are "slow" nodes in the cluster (with low loading speed) or if too much segments scheduled to be replicated to some particular node (faster loading could be preferred to better segments distribution). Desired value depends on segments loading speed, acceptable replication time and number of nodes. Value 1000 could be a start point for a rather big cluster. Default value is 100. |100|
+|`maxSegmentsInNodeLoadingQueue`|The maximum number of segments that could be queued for loading to any given server. This parameter could be used to speed up segments loading process, especially if there are "slow" nodes in the cluster (with low loading speed) or if too much segments scheduled to be replicated to some particular node (faster loading could be preferred to better segments distribution). Desired value depends on segments loading speed, acceptable replication time and number of nodes. Value 1000 could be a start point for a rather big cluster. |500|
```
Comment (ektravel) [inline docs/configuration/index.md:963]:
```suggestion
|`maxSegmentsInNodeLoadingQueue`|The maximum number of segments allowed in a loading queue for any given server. Use this parameter to load the segments faster&mdash;for example, if the cluster contains slow-loading nodes, or if there are too many segments to be replicated to a particular node (when faster loading is preferred to better segments distribution). Desired value depends on the loading speed of segments, acceptable replication time, and number of nodes. Value 1000 is a good starting point for a big cluster. |500|
```

**Review by ektravel — COMMENTED** [review]
Added some suggestions to improve readability.

**Comment by kfaraz** [comment]
Thanks for the review, @ektravel ! I have incorporated your feedback.

**Review by ektravel — APPROVED** [review]
Looks good from the docs standpoint.

**Pull request merged**